starcheck/state_checks.py

```diff
@@ -85,75 +107,78 @@ def get_obs_man_angle(npnt_tstart, backstop_file, default_large_angle=180):
+    ok = bs_cmds["event_type"] == "SCHEDULED_STOP_TIME"
     sched_stop = CxoTime(bs_dates[ok][0] if np.any(ok) else bs_dates[-1])
 
     # Get the states for available commands.  This automatically gets continuity.
-    return kadi_states.get_states(cmds=bs_cmds, start=rltt, stop=sched_stop,
-                                  state_keys=['pcad_mode'], merge_identical=True)
+    return kadi_states.get_states(
+        cmds=bs_cmds,
+        start=rltt,
+        stop=sched_stop,
+        state_keys=state_keys,
+        merge_identical=True,
+    )
 
 
 def ir_zone_ok(backstop_file, out=None):
+    """
+    Check that the high IR zone is all in NMM.
 
-    bs_cmds = kadi_commands.get_cmds_from_backstop(backstop_file)
-    bs_dates = bs_cmds['date']
-    ok = bs_cmds['event_type'] == 'RUNNING_LOAD_TERMINATION_TIME'
-    rltt = CxoTime(bs_dates[ok][0] if np.any(ok) else bs_dates[0])
-
-    # Scheduled stop time is the end of propagation, either the explicit
-    # time as a pseudo-command in the loads or the last backstop command time.
-    ok = bs_cmds['event_type'] == 'SCHEDULED_STOP_TIME'
-    sched_stop = CxoTime(bs_dates[ok][0] if np.any(ok) else bs_dates[-1])
+    :param backstop_file: backstop file
+    :param out: optional output file name (for plain text report)
+    :returns: True if high IR zone time is all in NMM, False otherwise
+    """
 
-    # Get the states for available commands.  This automatically gets continuity.
-    state_keys = ['pcad_mode', 'obsid']
-    states = kadi_commands.states.get_states(cmds=bs_cmds, start=rltt, stop=sched_stop,
-                                             state_keys=state_keys, merge_identical=True)
+    states = get_states(backstop_file, state_keys=["pcad_mode", "obsid"])
+    bs_cmds = kadi_commands.get_cmds_from_backstop(backstop_file)
 
-    perigee_cmds = bs_cmds[(bs_cmds['type'] == 'ORBPOINT')
-                           & (bs_cmds['event_type'] == 'EPERIGEE')]
+    perigee_cmds = bs_cmds[
+        (bs_cmds["type"] == "ORBPOINT") & (bs_cmds["event_type"] == "EPERIGEE")
+    ]
 
     all_ok = True
     out_text = ["IR ZONE CHECKING"]
-    for perigee_time in perigee_cmds['time']:
-
+    for perigee_time in perigee_cmds["time"]:
         # Current high ir zone is from 25 minutes before to 27 minutes after
+        # These don't seem worth vars, but make sure to change in the text below
+        # too when these are updated.
         ir_zone_start = perigee_time - 25 * 60
         ir_zone_stop = perigee_time + 27 * 60
+        out_text.append(f"Checking perigee at {CxoTime(perigee_time).date}")
         out_text.append(
-            f"Checking perigee at {CxoTime(perigee_time).date}")
-        out_text.append(
-            f"  High IR Zone from {CxoTime(ir_zone_start).date} to {CxoTime(ir_zone_stop).date}")
-        ok = (states['tstart'] <= ir_zone_stop) & (states['tstop'] >= ir_zone_start)
+            f"  High IR Zone from (perigee - 25m) {CxoTime(ir_zone_start).date} "
```

Comment:
There's a substantive change here in that I added the text to include the time in minutes.

Reply:
Is this getting warmer or do we need fancier conditional formatting?
```
IR ZONE CHECKING
Checking perigee at 2023:297:12:30:13.250
  High IR Zone from (perigee - 25m) 2023:297:12:05:13.250 to (perigee + 27m) 2023:297:12:57:13.250
  state 2023:297:11:59:00.000 (perigee + -31.2 min) 2023:297:12:51:40.251 (perigee + 21.5) NMAN
  state 2023:297:12:51:40.251 (perigee + 21.5 min) 2023:297:18:06:47.021 (perigee + 336.6) NPNT
Checking perigee at 2023:300:03:56:35.759
  High IR Zone from (perigee - 25m) 2023:300:03:31:35.759 to (perigee + 27m) 2023:300:04:23:35.759
  state 2023:300:03:26:00.000 (perigee + -30.6 min) 2023:300:04:18:40.251 (perigee + 22.1) NMAN
  state 2023:300:04:18:40.251 (perigee + 22.1 min) 2023:300:09:17:49.988 (perigee + 321.2) NPNT
```